lib/cartopy/mpl/solar.py

```diff
@@ -0,0 +1,124 @@
+# (C) British Crown Copyright 2018, Met Office
+#
+# This file is part of cartopy.
+#
+# cartopy is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cartopy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with cartopy.  If not, see <https://www.gnu.org/licenses/>.
+
+from __future__ import (absolute_import, division, print_function)
+import numpy as np
+
+
+def julian_day(date):
+    """
+    Calculate the Julian day from an input datetime.
+
+    Parameters
+    ----------
+    date
+        A UTC datetime object.
+
+    Note
+    ----
+    Algorithm implemented following equations from Chapter 3 (Algorithm 14):
+    Vallado, David 'Fundamentals of Astrodynamics and Applications', (2007)
+
+    """
+    year = date.year
+    month = date.month
+    day = date.day
+    hour = date.hour
+    minute = date.minute
+    second = date.second
+
+    # January/February correspond to months 13/14 respectively
+    # for the constants to work out properly
+    if month < 3:
+        month += 12
+        year -= 1
+
+    B = 2 - int(year/100) + int(int(year/100)/4)
+    C = ((second/60 + minute)/60 + hour)/24
+
+    JD = (int(365.25*(year + 4716)) + int(30.6001*(month+1)) +
+          day + B - 1524.5 + C)
+    return JD
+
+
+def solar_position(date):
+    """
+    Calculate the latitude and longitude point where the sun is
+    directly overhead for the given date.
+
+    Parameters
+    ----------
+    date
+        A UTC datetime object.
+
+    Returns
+    -------
+    (latitude, longitude) in degrees
+
+    Note
+    ----
+    Algorithm implemented following equations from Chapter 5 (Algorithm 29):
+    Vallado, David 'Fundamentals of Astrodynamics and Applications', (2007)
+
+    """
+    # Constants are in degrees in the book, so we need to convert the values
+    # when taking sin/cos
+    deg2rad = np.pi/180.
+    # Centuries from J2000
+    T_UT1 = (julian_day(date) - 2451545.0)/36525
+
+    # solar longitude (deg)
+    lambda_M_sun = (280.460 + 36000.771*T_UT1) % 360
+
+    # solar anomaly (deg)
+    M_sun = (357.5277233 + 35999.05034*T_UT1) % 360
+
+    # ecliptic longitude
+    lambda_ecliptic = (lambda_M_sun + 1.914666471*np.sin(M_sun*deg2rad) +
+                       0.019994643*np.sin(2*M_sun*deg2rad))
+
+    # obliquity of the ecliptic (epsilon in Vallado's notation)
+    epsilon = 23.439291 - 0.0130042*T_UT1
+
+    # declination of the sun
+    delta_sun = np.arcsin(np.sin(epsilon*deg2rad) *
+                          np.sin(lambda_ecliptic*deg2rad))/deg2rad
```

Comment:
You do have comment above each calculation though, so you could just annotate with the units (as is already done a couple times.)

Reply:
Yes, I could also change all of the constants to radians so we wouldn't even need to call deg2rad at all. This is not going to be the bottleneck here, so I think maintaining consistency with the textbook so that someone can reference where the code is obtained from is worth something here.